src/core/optimization_container.jl

```diff
@@ -289,6 +289,10 @@ function get_problem_size(container::OptimizationContainer)
     return "The current total number of variables is $(vars) and total number of constraints is $(cons)"
 end
 
+abstract type ConstructStage end
+struct ArgumentConstructStage end
+struct ConstraintConstructStage end
```

Comment:
This Stage is also building the objective function. Should we make a separate stage for that or rename the stage?

Reply:
`ModelConstructStage`